tools/tenscan/frontend/src/components/modules/batches/batch-height-details.tsx

```diff
@@ -0,0 +1,196 @@
+import JSONPretty from "react-json-pretty";
+import { useState } from "react";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "../../ui/tooltip";
+
+export function BatchHeightDetailsComponent({
+  batchDetails,
+}: {
+  batchDetails: Batch;
+}) {
+  const { decryptedRollup, decryptEncryptedData } = useRollupsService();
+  const [showDecryptedData, setShowDecryptedData] = useState(false);
+
+  return (
+    <div className="space-y-8">
+      <KeyValueList>
+        <KeyValueItem
+          label="Batch Height"
+          value={"#" + Number(batchDetails?.height)}
+        />
+        <KeyValueItem
+          label="Sequence"
+          value={"#" + Number(batchDetails?.sequence)}
+        />
+        <KeyValueItem
+          label="Hash"
+          value={
+            <Link
+              href={`/batch/${batchDetails?.header?.hash}`}
+              className="text-primary"
+            >
+              <TruncatedAddress address={batchDetails?.header?.hash} />
+            </Link>
+          }
+        />
+        <KeyValueItem
+          label="Parent Hash"
+          value={
+            <TruncatedAddress address={batchDetails?.header?.parentHash} />
+          }
+        />
+        <KeyValueItem
+          label="State Root"
+          value={<TruncatedAddress address={batchDetails?.header?.stateRoot} />}
+        />
+        <KeyValueItem
+          label="Transactions Root"
+          value={
+            <TruncatedAddress
+              address={batchDetails?.header?.transactionsRoot}
+            />
+          }
+        />
+        <KeyValueItem
+          label="Receipts Root"
+          value={
+            <TruncatedAddress address={batchDetails?.header?.receiptsRoot} />
+          }
+        />
+        <KeyValueItem
+          label="Timestamp"
+          value={
+            <Badge variant={"secondary"}>
+              {formatTimeAgo(batchDetails?.header?.timestamp)}
+            </Badge>
+          }
+        />
+        <KeyValueItem
+          label="L1 Proof"
+          value={<TruncatedAddress address={batchDetails?.header?.l1Proof} />}
+        />
+        <KeyValueItem
+          label="Gas Limit"
+          value={formatNumber(batchDetails?.header?.gasLimit)}
+        />
+        <KeyValueItem
+          label="Gas Used"
+          value={formatNumber(batchDetails?.header?.gasUsed)}
+        />
+        <KeyValueItem
+          label="Base Fee Per Gas"
+          value={batchDetails?.header?.baseFeePerGas || "-"}
+        />
+        <KeyValueItem
+          label="Inbound Cross Chain Hash"
+          value={
+            <TruncatedAddress
+              address={batchDetails?.header?.inboundCrossChainHash}
+            />
+          }
+        />
+        <KeyValueItem
+          label="Inbound Cross Chain Height"
+          value={Number(batchDetails?.header?.inboundCrossChainHeight)}
+        />
+        <KeyValueItem
+          label="Transfers Tree"
```

Comment:
I think we can probbaly remove this as well from the UI - but maybe worth checking with Cais.